### PR TITLE
add docs for #map_with_index

### PR DESCRIPTION
**Pull request by wontruefree**
adds an example to Array#map_with_index

### CODE REVIEW

File: src/array.cr [hunk 1]
```diff
@@ -1216,6 +1216,12 @@ class Array(T)
   #
   # Accepts an optional *offset* parameter, which tells it to start counting
   # from there.
+  #
+  # ```
+  # gems = [:crystal, :pearl, :diamond]
```
Comment (straight-shoota) [inline src/array.cr:1221]: Let's make both examples as similar as possible.
```suggestion
  # gems = ["crystal", "pearl", "diamond"]
```

**Review by straight-shoota — APPROVED** [review]

**Review by bcardiff — APPROVED** [review]

File: src/array.cr [hunk 2]
```diff
@@ -1216,6 +1216,12 @@ class Array(T)
   #
   # Accepts an optional *offset* parameter, which tells it to start counting
   # from there.
+  #
+  # ```
+  # gems = ["crystal", "pearl", "diamond"]
+  # results = gems.map_with_index { |e, i| "#{i}: #{e}" }
```
Comment (Sija) [inline src/array.cr:1222]:
```suggestion
  # results = gems.map_with_index { |gem, idx| "#{idx}: #{gem}" }
```

File: src/array.cr [hunk 3]
```diff
@@ -1224,6 +1230,12 @@ class Array(T)
   #
   # Accepts an optional *offset* parameter, which tells it to start counting
   # from there.
+  #
+  # ```
+  # gems = ["crystal", "pearl", "diamond"]
+  # gems.map_with_index! { |e, i| "#{i}: #{e}" }
```
Comment (Sija) [inline src/array.cr:1236]:
```suggestion
  # gems.map_with_index! { |gem, idx| "#{idx}: #{gem}" }
```

File: src/array.cr (hunk 2 above)
Comment (straight-shoota) [reply src/array.cr:1222]: I think `i` is fine - and `idx` doesn't have any noticable benefit. But `gem` is a good idea.

Comment (wontruefree) [reply src/array.cr]: I updated `gem` but not `idx`.  I think that is fine but let me all know what you think.

**Review by straight-shoota — APPROVED** [review]

**Review by sdogruyol — APPROVED** [review]
Thank you @wontruefree

**Comment by straight-shoota** [comment]
Thank you @wontruefree

**Pull request merged**